6tempcontrol

```diff
@@ -292,26 +308,32 @@ do
     fi
 
     POWERLIMIT=${PWRLIMIT%%.*}
-    TEMP_DIFF=$((${TARGET_TEMP[${GPU}]} - $CURRENT_TEMP))
+    TEMP_DIFF=$((${TARGET_TEMP[${GPU}]} - CURRENT_TEMP))
     NEW_FAN_SPEED=$CURRENT_FAN
+    HIGH_TEMP_DIFF=$(( ALLOWED_TEMP_DIFF * 2 ))
 
     echo -e "${B}GPU $GPU${N}, Target temp: ${B}${TARGET_TEMP[${GPU}]}${N}, Current: ${B}$CURRENT_TEMP${N}, Diff: ${B}$TEMP_DIFF${N}, Fan: ${B}$CURRENT_FAN${N}, Power: ${B}$POWERDRAW${N}"
 
-    if [ "$CURRENT_TEMP" -gt "${TARGET_TEMP[${GPU}]}" ]; then
-      # This can be far more advanced. For now if difference is more than 5 C multiply adjustement by 2
-      if [ $TEMP_DIFF -lt "-5" ]; then
-        FAN_ADJUST_CALCULATED=$(($FAN_ADJUST * 2))
+    if (( CURRENT_TEMP > "${TARGET_TEMP[${GPU}]}" ))
```

Comment:
Here maybe you need to remove quotes because otherwise could be parsed as string, I don't know how it behaves. Just change to TARGET_TEMP[GPU]